ios/KMMICKIFUITests/Sources/Specs/Presentation/Module/ThankScreenSpec.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  ThankScreenSpec.swift
+//  KMMICKIFUITests
+//
+//  Created by MarkG on 18/05/2023.
+//  Copyright © 2023 Nimble. All rights reserved.
+//
+
+import Factory
+import Nimble
+import Quick
+import Shared
+
+@testable import KMMIC
+
+final class ThankScreenSpec: QuickSpec {
+
+    override func spec() {
+
+        describe("a Thank screen") {
+
+            var navigator: NavigatorMock!
+            var didSetScreen = false
+
+            beforeEach {
+                guard !didSetScreen else { return }
```

Comment:
should we assert it as a failure in this else block

Reply:
@phongvhd93 I aim to add this check to make sure it only runs once, it similar to `beforeAll()` but Quick doesn't have this function 🙏

Reply:
Optional:
You can use something like `fail("")`. This is the [source code](https://github.com/Quick/Nimble/blob/d0bea4ff70428c0fb30d26c8c1fa2500cd4570ff/Sources/Nimble/DSL.swift#L21-L35)